community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/steps/pickBestPlanUsingHintsAndCost.scala

```diff
@@ -26,7 +26,7 @@ object pickBestPlanUsingHintsAndCost extends LogicalPlanningFunction0[CandidateS
   private final val VERBOSE = false
 
   override def apply(implicit context: LogicalPlanningContext): CandidateSelector = new CandidateSelector {
-    override def apply[X](projector: (X) => LogicalPlan, input: Iterable[X]): Option[X] = {
+    override def apply[X](projector: (X) => LogicalPlan, input: Iterator[X]): Option[X] = {
```

Comment:
didn't profile but this goes down to the core loop of IDP. Basically pickBest is computing a minimum which can be done in a streaming fashion, i.e. doesn't need to build a seq upfront. If this interface would take an iterable, every time we generate candidates in idp we'd need to put them in seq but keeping this an iterator this can be avoided. The number of candidates can be potentially large and thus memory wasteful. I haven't profiled this so we could either change it on this PR or have a separate PR that changes it back and if we don't see an impact in iscyfayet keep otherwise revert.